dev/src/index.ts

```diff
@@ -505,23 +505,20 @@ export class Firestore implements firestore.Firestore {
     if (settings.host !== undefined) {
       validateHost('settings.host', settings.host);
       if (
-        settings.servicePath !== undefined &&
-        settings.servicePath !== settings.host
+        (settings.servicePath !== undefined &&
+          settings.servicePath !== settings.host) ||
+        (settings.apiEndpoint !== undefined &&
+          settings.apiEndpoint !== settings.host)
```

Comment:
Host is "hostname:port". At least servicePath is just hostname. Can you verify that this works as expected?

Reply:
Changed logic to compare hostnames.